File: leetcode/algorithms/sort_colors/sort_colors.cc
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int colors[3] = {0};

        for (int i = 0; i < nums.size(); ++i) {
            ++colors[nums[i]];
        }

        int position = 0;

        for (int i = 0; i < 3; ++i) {
            for (int j = 0; j < colors[i]; ++j) {
                nums[position++] = i;
            }
        }
    }
};

File: leetcode/algorithms/sort_colors/sort_colors.py
class Solution(object):
    def sortColors(self, nums):
        colors = [0] * 3

        for num in nums:
            colors[num] += 1

        position = 0

        for i in xrange(3):
            for j in xrange(colors[i]):
                nums[position] = i
                position += 1
